play-json-ops-common/src/main/scala/play/api/libs/json/ops/FormatOps.scala

```diff
@@ -8,6 +8,24 @@ import scala.util.control.NonFatal
 
 object FormatOps {
 
+  /**
+   * Extracts the class name of the [[Enumeration]] removing any `$` symbols without throwing an exception.
+   *
+   * @note this will include any outer object or class names separated by `.`s
+   */
+  def enumClassName(o: Enumeration): String = {
+    // This logic is designed to be robust without much noise
+    // 1. use getName to avoid runtime exceptions from getSimpleName
+    // 2. filter out '$' anonymous class / method separators
+    // 3. start the full class name from the first upper-cased outer class name
+    //    (to avoid picking up unnecessary package names)
+    o.getClass.getName
+      .split('.')
+      .last
```

Comment:
```suggestion
      .lastOption
```

Because 
```
Seq.empty.last
java.util.NoSuchElementException: last of empty list
```

Reply:
Hmm.  Is this a realistic concern?  It looks like `.last` could only fail if `o.getClass.getName` returned an empty String, and I don't know if that's possible.  This might be a legit use of `.last`.

Reply:
I can put Justin's comment in the code. That was my thinking as well.